InventoryCatalog/Model/IsProductSalable.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Model;
+
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\Product\Attribute\Source\Status;
+use Magento\InventorySalesApi\Api\AreProductsSalableInterface;
+
+/**
+ * Get salable product status service.
+ */
+class IsProductSalable
+{
+    /**
+     * @var GetStockIdForCurrentWebsite
+     */
+    private $getStockIdForCurrentWebsite;
+
+    /**
+     * @var AreProductsSalableInterface
+     */
+    private $areProductsSalable;
+
+    /**
+     * @var array
+     */
+    private $productStatusCache;
+
+    /**
+     * @param GetStockIdForCurrentWebsite $getStockIdForCurrentWebsite
+     * @param AreProductsSalableInterface $areProductsSalable
+     */
+    public function __construct(
+        GetStockIdForCurrentWebsite $getStockIdForCurrentWebsite,
+        AreProductsSalableInterface $areProductsSalable
+    ) {
+        $this->getStockIdForCurrentWebsite = $getStockIdForCurrentWebsite;
+        $this->areProductsSalable = $areProductsSalable;
+    }
+
+    /**
+     * Verify product salable status.
+     *
+     * @param Product $product
+     * @return bool
+     */
+    public function execute(Product $product): bool
```

Comment:
this violates modularity we should not accept and work with product model in inventory modules